algebra/tests/matrix/matrix_multiplication.c

```diff
@@ -303,7 +313,106 @@ TEST(group_matrix_prod_bigMat, matrix_prod_bigMatsAllMatTrp)
 }
 
 
-/* FIXME: Add tests for incorrect matrix sizes to multiply */
+/* This tests checks if function changes source matrices after success */
+TEST(group_matrix_prod_bigMat, matrix_prod_sourceRetain)
+{
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, algebraTests_matrixCopy(&M4, &M1));
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, algebraTests_matrixCopy(&M5, &M2));
+
+	TEST_ASSERT_EQUAL_INT(PRODUCT_OK, matrix_prod(&M1, &M2, &M3));
+
+	TEST_ASSERT_EQUAL_INT(CHECK_OK, algebraTest_equalMatrix(&M1, &M4));
+	TEST_ASSERT_EQUAL_INT(CHECK_OK, algebraTest_equalMatrix(&M2, &M5));
+}
+
+
+TEST_GROUP(group_matrix_prod_incorrMats);
```

Comment:
What do you think? Sounds more natural. May be substituted in all `incorr` occurrences
```suggestion
TEST_GROUP(group_matrix_prod_badMats);
```